### PR TITLE
Issue #1777: Advanced link options

### PR DESCRIPTION
**Pull request by klonos**
https://github.com/backdrop/backdrop-issues/issues/1777

### CODE REVIEW

File: core/modules/filter/filter.pages.inc [hunk 1]
```diff
@@ -373,7 +373,7 @@ function filter_format_editor_link_form($form, &$form_state, $format) {
     ),
     '#weight' => -10,
     '#autocomplete_path' => 'filter-link/autocomplete',
-    '#description' => t('Enter a known path, or start typing to get suggestions'),
+    '#description' => t('Enter a link to an external page or a known internal path, or start typing to get suggestions for paths of existing content.'),
```
Comment (jenlampton) [inline core/modules/filter/filter.pages.inc]: Can we make this two sentences? `Enter a link to an external page, or a known internal path. Start typing to get suggestions for paths to existing content.`

File: core/modules/filter/filter.pages.inc [hunk 2]
```diff
@@ -432,22 +432,54 @@ function filter_format_editor_link_form($form, &$form_state, $format) {
...
+  $form['advanced']['id'] = array(
+    '#title' => t('ID'),
+    '#description' => t('Allows linking to this content using a URL !fragment. Must be unique.', array('!fragment' => $fragment_info_link)),
+    '#type' => 'textfield',
+    '#default_value' => isset($values['id']) ? $values['id'] : '',
+    '#parents' => array('attributes', 'id'),
+  );
+
+  $form['advanced']['rel'] = array(
+    '#title' => t('Relation (rel)'),
```
Comment (jenlampton) [inline core/modules/filter/filter.pages.inc]: I think this is "Relationship" rather than "Relation" ?

File: core/modules/filter/filter.pages.inc [hunk 3]
```diff
@@ -432,22 +432,54 @@ function filter_format_editor_link_form($form, &$form_state, $format) {
...
+    '#description' => t('Often used by JavaScript libraries.'),
```
Comment (jenlampton) [inline core/modules/filter/filter.pages.inc]: I'm not sure this description adds much value. Can we remove it, or replace it with an explanation of Rel? 'Specifies the relationship between the current document and the linked document.'

File: core/modules/filter/filter.pages.inc [hunk 4]
```diff
@@ -432,22 +432,54 @@ function filter_format_editor_link_form($form, &$form_state, $format) {
...
+    '#type' => 'textfield',
+    '#default_value' => isset($values['rel']) ? $values['rel'] : '',
+    '#parents' => array('attributes', 'rel'),
+  );
+
+  $form['advanced']['title'] = array(
+    '#title' => t('Link title'),
+    '#description' => t('Populates the title attribute of the link. Usually shown as a small tooltip on hover.'),
```
Comment (jenlampton) [inline core/modules/filter/filter.pages.inc]: I'm not sure about 'Populates' how about... `The title attribute for the link. Usually shown as a small tooltip on hover.`

File: core/modules/filter/filter.pages.inc (hunk 1 above)
Comment (klonos) [reply core/modules/filter/filter.pages.inc]: 👍 ...I have gone with `Enter a link to an external page or a known internal path. Alternatively, start typing to get suggestions for paths of existing content.`

File: core/modules/filter/filter.pages.inc (hunk 2 above)
Comment (klonos) [reply core/modules/filter/filter.pages.inc]: 👍

File: core/modules/filter/filter.pages.inc (hunk 3 above)
Comment (klonos) [reply core/modules/filter/filter.pages.inc]: Yep, that's what https://www.w3schools.com/tags/att_a_rel.asp says too 😉

Comment (klonos) [reply core/modules/filter/filter.pages.inc]: ...but have used "page" instead of "document".

File: core/modules/filter/filter.pages.inc (hunk 4 above)
Comment (klonos) [reply core/modules/filter/filter.pages.inc]: 👍

File: core/modules/filter/filter.pages.inc [hunk 5]
```diff
@@ -432,22 +432,54 @@ function filter_format_editor_link_form($form, &$form_state, $format) {
     '#parents' => array('attributes', 'text'),
   );
 
-  $form['target'] = array(
+  $form['advanced'] = array(
+    '#title' => t('Advanced options'),
```
Comment (jenlampton) [inline core/modules/filter/filter.pages.inc:436]: I really don't like calling them 'Advanced'. They are just more options. Perhaps less-common, but not advanced.

File: core/modules/filter/filter.pages.inc [hunk 6]
```diff
@@ -432,22 +432,54 @@ function filter_format_editor_link_form($form, &$form_state, $format) {
...
+    '#type' => 'fieldset',
+    '#collapsible' => TRUE,
+    '#collapsed' => TRUE,
+  );
+
+  $form['advanced']['target'] = array(
     '#title' => t('Open in new window'),
```
Comment (jenlampton) [inline core/modules/filter/filter.pages.inc:443]: Open what? Can we change this to `Open link destination in new window`.

File: core/modules/filter/filter.pages.inc [hunk 7]
```diff
@@ -432,22 +432,54 @@ function filter_format_editor_link_form($form, &$form_state, $format) {
...
+  $fragment_info_link = l(t('fragment'), 'https://en.wikipedia.org/wiki/Fragment_identifier', array('attributes' => array('target' => '_blank')));
+  $form['advanced']['id'] = array(
+    '#title' => t('ID'),
+    '#description' => t('Allows linking to this content using a URL !fragment. Must be unique.', array('!fragment' => $fragment_info_link)),
+    '#type' => 'textfield',
+    '#default_value' => isset($values['id']) ? $values['id'] : '',
+    '#parents' => array('attributes', 'id'),
+  );
+
+  $form['advanced']['rel'] = array(
```
Comment (jenlampton) [inline core/modules/filter/filter.pages.inc:467]: Can we swap `rel` and `title` so that `title` comes first? I think `rel` is less commonly used.

**Comment by jenlampton** [comment]
Replaced by https://github.com/backdrop/backdrop/pull/2638

**Pull request closed without merging**